docs/contributor/updateability.md

```diff
@@ -0,0 +1,349 @@
+# Upgrading MicroShift
+
+Following document is meant to be a contributor's introduction
+to the feature of MicroShift's updateability.
+
+## On-disk artifacts
+
+New on-disk artifacts were added:
+- `/var/lib/microshift-backups/`
+    - `health.json`
+    - backup directories
+- `/var/lib/microshift/version`
+
+### `/var/lib/microshift-backups/`
+
+New, additional directory for keeping health and backup related artifacts.
+For ostree systems it's a non-configurable directory where backups are created.
+For non-ostree systems it's a default (configurable) directory for backups.
+
+#### `health.json` [ostree-only]
+
+Contains information about health of the system.
+It's updated by green and red script supplied by MicroShift.
+
+Because MicroShift's healthcheck takes some time to assess the state of the MicroShift,
+this file might be updated after couple minutes into system boot.
+
+Schema:
+```json
+{
+    "health": "",
+    "deployment_id": "",
+    "boot_id": ""
+}
+```
+
+- `health`: System status according to the greenboot healthchecks.
+  - Expected values: `healthy` or `unhealthy`
+- `deployment_id`: OSTree deployment ID.
+  - Obtained with a command: `rpm-ostree status --json | jq '.deployments[].id'`
+- `boot_id`: Boot ID generated by kernel.
+  - Obtained with command: `tr -d '-' < /proc/sys/kernel/random/boot_id`.
+  - Hyphens are removed to match format used by `journalctl` (see `journalctl --list-boots --reverse`)
+
+
+#### Backup directories
+
+> Note: Only ostree-based systems perform automated backups and restores.
+
+`/var/lib/microshift-backups/` also stores backups of MicroShift data.
+Backups are created by copying data directory with `--reflink`
+option to leverage Copy-on-Write functionality.
+
+Naming schema of backup directories: `deployment-id_boot-id[_unhealthy]`.
+
+Suffix `_unhealthy` is used to mark backups of unhealthy systems.
+These backups are performed when all existing metadata suggests that it is expected
+that MicroShift should wipe the data and start from clean state
+(for example "FDO" scenario where unhealthy deployment leaves stale data).
+
+### `/var/lib/microshift/version`
+
+TODO
+
+Schema:
+```
+MAJOR.MINOR.PATCH
+```
+
+All of MAJOR, MINOR, and PATCH are unsigned integers.
+For example: `4.14.0`
+
+## Greenboot [ostree-only]
```

Comment:
Same comment, the greenboot scripts can and might be installed on systems not based on ostree.

Reply:
Well, yes. But it's usefulness is greatly reduced. Also, you can perform many lifecycle operations on microshift on RPM without rerunning the greenboot.

I'll add a comment